pkg/admission/admission.go

```diff
@@ -154,6 +154,23 @@ func (a *ParallelAdmission) ValidateResources(ctx context.Context, localResource
 			Name:      objName,
 		}
 
+		// If the object includes a pod spec (i.e. Deployment), create a Pod object out of it for validation.
+		// For admission only pod resources will be enforced, deployments won't.
+		podExtractor := psadmission.DefaultPodSpecExtractor{}
+		if podExtractor.HasPodSpec(resource.GroupResource()) {
```

Comment:
this might be creating a pod from a pod but I suppose we can keep this generic handling